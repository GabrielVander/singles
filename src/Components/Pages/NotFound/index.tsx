import React, { ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import { HOME } from '../../../Routes/AppRoutes';
import { Box, Button, Heading, Image, Main } from 'grommet';
import notFound from '../../../Assets/page-not-found.svg';
import AppFooter from '../../Commom/AppFooter';

function NotFound(): ReactElement {
    const { t } = useTranslation(['notFound']);

    return (
        <>
            <Main align="center" justify="center">
                <Box direction="column" gap="large" align="center" justify="around" pad="medium">
                    <Box
                        align="center"
                        animation="pulse"
                        justify="center"
                        width="medium"
                        height="medium"
                        style={{ fontSize: '10vw' }}
                    >
                        <Image
                            fit="contain"
                            style={{ width: '100%', height: '100%' }}
                            src={notFound}
                            a11yTitle={t('notFound:illustrationAccessibilityLabel')}
                        />
                    </Box>
                    <Box gap="small">
                        <Heading level="3">{t('notFound:description')}</Heading>
                        <Box align="center">
                            <Link to={HOME.path}>
                                <Button primary label={t('notFound:buttonLabel')} />
                            </Link>
                        </Box>
                    </Box>
                </Box>
            </Main>
            <AppFooter />
        </>
    );
}

export default NotFound;
