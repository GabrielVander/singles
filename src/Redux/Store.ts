import {createStore} from "redux";
import rootReducer from "./Reducers";

class Store {
    static INITIAL_GLOBAL_STATE = {};
    // @ts-ignore
    static store = createStore(rootReducer, Store.INITIAL_GLOBAL_STATE, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
}

export default Store;
