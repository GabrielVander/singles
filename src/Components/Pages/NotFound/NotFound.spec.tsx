import { render } from '@testing-library/react';
import React from 'react';
import NotFound from './index';
import { Provider } from 'react-redux';
import Store from '../../../Redux/Store';
import { BrowserRouter } from 'react-router-dom';

// Jest's official workaround for the "window.matchMedia is not a function" error
beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation((query) => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // deprecated
            removeListener: jest.fn(), // deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });
});

test('matches snapshot', () => {
    const component = render(
        <Provider store={Store.store}>
            <BrowserRouter>
                <NotFound />
            </BrowserRouter>
        </Provider>,
    );

    expect(component).toMatchSnapshot();
});
