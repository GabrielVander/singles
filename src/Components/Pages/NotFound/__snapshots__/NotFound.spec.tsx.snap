// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="StyledBox-sc-13pk1d4-0 emVEPy"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 dSTygu"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bzOKJH"
            style="font-size: 10vw;"
          >
            <img
              aria-label="notFound:illustrationAccessibilityLabel"
              class="StyledImage-ey4zx9-0 dfToQT"
              src="page-not-found.svg"
              style="width: 100%; height: 100%;"
            />
          </div>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hleNBA"
          />
          <div
            class="StyledBox-sc-13pk1d4-0 gAsOLC"
          >
            <h3
              class="StyledHeading-sc-1rdh4aw-0 hBHJOo"
            >
              notFound:description
            </h3>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eaIbGj"
            />
            <div
              class="StyledBox-sc-13pk1d4-0 daweyS"
            >
              <a
                href="/home"
              >
                <button
                  class="StyledButton-sc-323bzc-0 eVPMSD"
                  type="button"
                >
                  notFound:buttonLabel
                </button>
              </a>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="StyledBox-sc-13pk1d4-0 emVEPy"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 dSTygu"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 bzOKJH"
          style="font-size: 10vw;"
        >
          <img
            aria-label="notFound:illustrationAccessibilityLabel"
            class="StyledImage-ey4zx9-0 dfToQT"
            src="page-not-found.svg"
            style="width: 100%; height: 100%;"
          />
        </div>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hleNBA"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 gAsOLC"
        >
          <h3
            class="StyledHeading-sc-1rdh4aw-0 hBHJOo"
          >
            notFound:description
          </h3>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eaIbGj"
          />
          <div
            class="StyledBox-sc-13pk1d4-0 daweyS"
          >
            <a
              href="/home"
            >
              <button
                class="StyledButton-sc-323bzc-0 eVPMSD"
                type="button"
              >
                notFound:buttonLabel
              </button>
            </a>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
